drivers/virtio/ring/virtio_ring.c

```diff
@@ -43,6 +43,7 @@
 #include <uk/plat/io.h>
 #include <virtio/virtio_ring.h>
 #include <virtio/virtqueue.h>
+#include <virtio/virtio_bus.h>
```

Comment:
Where do we actually use the declarations from `virtio_bus`?
I'm likely suffering a brainfart, but I can't find any in the PR diff.

Reply:
These are included from `Makefile.uk`:
```
# common virtio headers
LIBVIRTIO_RING_CINCLUDES-y  += -I$(UK_DRIV_LIBVIRTIO_BASE)/include
LIBVIRTIO_RING_CINCLUDES-y  += -I$(LIBVIRTIO_BUS_BASE)/include
LIBVIRTIO_RING_CINCLUDES-y  += -I$(LIBVIRTIO_RING_BASE)/include
```

Reply:
It's the feature defines (`VIRTIO_FEATURE_HAS`, etc.)
The header is possibly not the best place for them but they were moved into the bus library during the re-arch move.

Reply:
Yes these headers were moved as part of re-arch verbatim, as in general re-arch avoids refactoring to limit the scope of work. And since the different virtio components were split into individual libraries `virtio_bus.h` naturally became part of `libvirtio_bus`.

From what I see the reason `VIRTIO_FEATURE_*` are defined in `virtio_bus.h` is likely because the same header defines `virtio_confg_ops` (which among others also include `features_get` / `features_set`) and that is effectively part of `struct virtio_dev` which is defined there too.

On the other hand there is also `virtio_config.h`  as part of the common headers, so that could possible be a better place?

PS: Sorry for the previous hasty and irrelevant answer 🙏🏼

Reply:
Users of the feature test macros also have to access the `features` field of `virtio_device`. So they have to rely on the header.